### ISSUE TITLE
core feature 2: follow user

### ISSUE THREAD

**Issue by Alex-DSJ** [yida-li/SOEN341#6]
the second core feature required by our customer is the user can follow others so that their can track someone.

**Comment by kevinlizh1992** [comment]
At the picture posting page, there needs to be UI/backend to go fetch a picture file on your computer since it's a webapp, it could be a simple attach file style UI for sending emails

**Comment by yida-li** [comment]
good idea! what other user stories should we include for this feature?

**Comment by kevinlizh1992** [comment]
Once upload completes, there should be a hyperlink/image to redirect you to your own page or to news feed

**Comment by successsb89** [comment]
yes, I think so too. I also suggest one idea as well. If user exists, will show up as a rectangle/suggested, and user can click on it, which will redirect to that users page where a follow button can be found

**Comment by kevinlizh1992** [comment]
Once followed, user being followed will receive notification about who followed them

**Comment by Alex-DSJ** [comment]
good, but first of all, we need a search bar to allow users search by the username.

**Comment by Alex-DSJ** [comment]
specification for following users:

- [ ] this function should implement by adding the link of other user's profile to the drop-down menu on the admin's profile page;
- [ ] the information of the following users should be stored in mongoDB;
- [ ] [optional] if the following users post new pictures, the user should receive notification, detail to be discussed

**Comment by yida-li** [comment]
The follower page requires list of users, this issue need to be postponed until we can get the follower button set up
